react/client/src/contexts/RoutingContext.tsx

```diff
@@ -11,18 +13,61 @@ const RoutingContextProvider = ({
   children,
   history,
 }: RoutingProviderProps) => {
-  const url = history.location.pathname;
-  const pageNumber = Number(url.slice(url.indexOf('/form') + 6)) || 0;
+  const [formSteps, setFormSteps] = useState([FORM_STEPS.NONE]);
```

Comment:
Yeah, I'm not the happiest about this change because it's basically tracking the already existing history.
I wanted to handle the dynamic case coming up with the involvement step. Previously we tracked by page numbers but we wouldn't know how to go backwards if we didn't have all the involvement checkboxes checked. For example: if we only have the "job" and "parenting" checkboxes checked and only referenced using numbers, pressing back on "parenting" would lead us to "school" -- instead of the expecting "job"

Tracking them in an array `formSteps` array will eventually look like [..., "involvement", "job", 'parenting"] and we'll know how to go back.

Reply:
Yep that's a great solution to a potential issue I hadn't even thought of